packages/plugin/webpack/src/WebpackConfig.ts

```diff
@@ -32,10 +32,16 @@ export default class WebpackConfigGenerator {
     d('Config mode:', this.mode);
   }
 
-  resolveConfig(config: Configuration | string): Configuration {
+  async resolveConfig(config: Configuration | string): Promise<Configuration> {
     if (typeof config === 'string') {
       // eslint-disable-next-line @typescript-eslint/no-var-requires, import/no-dynamic-require, global-require
-      return require(path.resolve(this.projectDir, config)) as Configuration;
+      const result = require(path.resolve(this.projectDir, config));
+
+      if (typeof result === 'function') {
+        return (await result()) as Configuration;
```

Comment:
```suggestion
        return (await Promise.resolve(result())) as Configuration;
```

Reply:
I think it's harmless to await sync function.